src/mlpack/core/data/string_algorithms.hpp

```diff
@@ -0,0 +1,109 @@
+/** 
+ * @file core/data/string_algorithms.hpp
+ * @author Gopi M. Tatiraju
+ *
+ * Utility fucntions related to string manipulation
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+#define MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+
+namespace mlpack{
+namespace data{
+
+/**
+ * A simple trim fucntion to strip off whitespaces 
+ * from both the side of string. If input is a string
+ * with all spaces then str will be empty string.
+ *
+ * @param str string to be trimmed
+ */
+inline void trim(std::string& str)
+{
+  /*if(str.size() < 2)
+  {
+    return;
+  }*/
+
+  if(str.find_first_not_of(' ') == std::string::npos)
+  {
+    str = "";
+    return;
+  }
+  size_t startIndex = 0;
+
+  while(std::isspace(str[startIndex]))
+  {
+    startIndex++;
+  }
+
+  size_t endIndex = str.size() - 1;
+
+  while(std::isspace(str[endIndex]))
+  {
+    endIndex--;
+  }
+
+  std::string trimmedStr = (endIndex - startIndex == str.size()) ? 
+                             std::move(str) : str.substr(startIndex, endIndex - startIndex + 1);
+
+  str = trimmedStr;
+}
+
+/**
+ * Trim off characters from start and end of
+ * of the string. The supplied fucntion is
+ * used to determine which characters will
+ * be trimmed off.
+ *
+ * @param str string to be trimmed
+ * @param func fucntion to determine the characters which should be trimmed
+ */
+inline void trim_if(std::string &str, bool (*func)(char))
+{
```

Comment:
Great work, I did not think you have already finished this one, perfect. Please do not consider my above comment considering this `trim_if`

Reply:
Also, you can use `std::function<bool(char)>` instead of old C style callback, I am not sure of the syntax, but it should work fine in both cases. :+1: